idea-plugin/src/main/kotlin/arrow/meta/dsl/ide/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -13,6 +13,22 @@ import com.intellij.psi.PsiElement
 import javax.swing.Icon
 
 interface LineMarkerSyntax {
+  /**
+   * Adds a fast line provider
+   * See [com.intellij.codeInsight.daemon.LineMarkerProvider] for notes how to implement a fast provider.
+   */
+  fun IdeMetaPlugin.addFastLineMarkerProvider(
```

Comment:
This doesn't add any LineMarkers to the Editor.

Reply:
It does, at least here. It's not displaying icons during indexing because our providers are not dumb-aware yet. Implementing the interface `PossiblyDumbAware` is probably best here as it allows to control the value.

Reply:
https://github.com/JetBrains/intellij-community/blob/192/platform/lang-impl/src/com/intellij/codeInsight/daemon/impl/LineMarkersPass.java#L93 is where the collectors are collected, `getMarkerProviders` filters by dumb-awareness.

Reply:
I see your point now. We should have the "fast" option global for everyone. Let me redefine that.

Thanks for looking into it :)